### PR TITLE
fix(styles): Style cleanup for Table Component

### PR DESCRIPTION
**Pull request by aberikashvili**
## Related Issue
Closes SAP/fundamental-styles#2726

## Description
This PR delivers, styling cleanup changes from fundamental-ngx repo

#### Please check whether the PR fulfills the following requirements

1. The output matches the design specs
- [x] All values are in `rem`
- [N/A] Text elements follow the truncation rules
- [N/A] hover state of the element follow design spec
- [N/A] focus state of the element follow design spec
- [N/A] active state of the element follow design spec
- [N/A] selected state of the element follow design spec
- [N/A] selected hover state of the element follow design spec
- [N/A] pressed state of the element follow design spec
- [N/A] Responsiveness rules - the component has modifier classes for all breakpoints
- [N/A] Includes Compact/Cosy/Tablet design
- [x] RTL support
2. The code follows fundamental-styles code standards and style
- [x] only one top level `fd-*` class is used in the file
- [x] BEM naming convention is used
- [x] Mixins are used for repeatable code (`fd-rtl`, `fd-ellipsis`, `fd-flex`, `fd-selected`, `fd-focus`, ect.)
- [x] A11y support - keyboard support, screenreader support, proper ARIA attributes, etc.
- [N/A] `fd-reset()` mixin is applied to all elements
- [x] Variables are used, if some value is used more than twice.
- [N/A] Checked if current components can be reused, instead of having new code.
3. Testing
- [N/A] tested Storybook examples with "CSS Resources" `normalize` option 
- [N/A] tested Storybook examples with "CSS Resources" `unnormalize` option 
- [N/A] Verified all styles in IE11
- [N/A] Updated tests
- [x] last commit message should have `[ci visual]` so it can trigger chromatic visual regression (e.g. `test: run chromatic visual regression [ci visual]`)
4. Documentation
- [N/A] Storybook documentation has been created/updated
- [N/A] Breaking Changes [wiki](https://github.com/SAP/fundamental-styles/wiki/Breaking-Changes) has been updated in case of breaking changes.

### CODE REVIEW

**Review by droshev — CHANGES_REQUESTED** [review]
@aberikashvili do we have examples showing those changes?

File: src/styles/table.scss [hunk 1]
```diff
@@ -11,6 +11,9 @@
 */
 $block: #{$fd-namespace}-table;
 
+$fd-table-border: 0.0625rem solid #e4e4e4;
```
Comment (InnaAtanasova) [inline src/styles/table.scss:14]: You don't need this part, it's a fallback generated with a script to make sure variables work in all browsers. 
All you need is:
```
$fd-table-border: var(--sapList_BorderWidth) solid var(--sapList_BorderColor);
```

File: src/styles/table.scss [hunk 2]
```diff
@@ -11,6 +11,9 @@
...
+$fd-table-border-var: var(--sapList_BorderWidth, 0.0625rem) solid var(--sapList_BorderColor, #e4e4e4);
```
Comment (InnaAtanasova) [inline src/styles/table.scss:15]: see comment above

File: src/styles/table.scss [hunk 3]
```diff
@@ -137,6 +140,39 @@ $block: #{$fd-namespace}-table;
 
   /****** Elements ******/
 
+  &__empty {
+    &-table-message {
+      padding: 1rem;
+      text-align: center;
+      display: flex;
+      align-items: center;
+      justify-content: center;
```
Comment (InnaAtanasova) [inline src/styles/table.scss:149]: There's a mixin you can use for this block of code. Please check `fd-flex-center` mixin.

File: src/styles/table.scss [hunk 4]
```diff
@@ -137,6 +140,39 @@ $block: #{$fd-namespace}-table;
...
+      align-content: center;
+      width: 100%;
+    }
+
+    &:not(.fixed-height) .fd-table__row:last-of-type {
```
Comment (InnaAtanasova) [inline src/styles/table.scss:154]: I don't know where the class `.fixed-height` is coming from but it doesn't have the correct naming. It should follow BEM and be prefixed with `fd-`

File: src/styles/table.scss [hunk 5]
```diff
@@ -137,6 +140,39 @@ $block: #{$fd-namespace}-table;
...
+      .fd-table__cell {
```
Comment (InnaAtanasova) [inline src/styles/table.scss:155]: You can't use `.fd-tabele__something`, you need to use `.#{$block}__something`.

File: src/styles/table.scss [hunk 6]
```diff
@@ -137,6 +140,39 @@ $block: #{$fd-namespace}-table;
...
+        border-bottom: 0;
+      }
+    }
+
+    .fd-table__body {
+      border-left: none;
+
+      @include fd-rtl() {
+        border-left: $fd-table-border;
+        border-left: $fd-table-border-var;
```
Comment (InnaAtanasova) [inline src/styles/table.scss:165]: The second line is not needed.

File: src/styles/table.scss [hunk 7]
```diff
@@ -137,6 +140,39 @@ $block: #{$fd-namespace}-table;
...
+        border-right: none;
+      }
+    }
+
+    &.fixed-height {
+      background-color: #fff;
+      background-color: var(--sapList_Background, #fff);
```
Comment (InnaAtanasova) [inline src/styles/table.scss:172]: First line is not needed, the second doesn't need the fallback. You can just have: background-color: var(--sapList_Background);

File: src/styles/table.scss [hunk 8]
```diff
@@ -240,6 +276,26 @@ $block: #{$fd-namespace}-table;
       @include fd-table-reset-spacing();
     }
 
+    &__group {
+      min-width: 100%;
+      border-left: none;
+
+      /**
+        Due to the changed table-layout group row cell may looe it's top border.
```
Comment (InnaAtanasova) [inline src/styles/table.scss:284]: Typo in the word `looe`

File: src/styles/table.scss [hunk 9]
```diff
@@ -240,6 +276,26 @@ $block: #{$fd-namespace}-table;
...
+        For example when previous (relatively to the group row) row columns tolal width < 100% of the table.
+      */
+      &::before {
+        content: '';
+        display: block;
+        position: absolute;
+        top: -0.0625rem;
+        left: 0;
+        right: 0;
```
Comment (InnaAtanasova) [inline src/styles/table.scss:293]: Shouldn't you handle RTL?

File: src/styles/table.scss [hunk 10]
```diff
@@ -240,6 +276,26 @@ $block: #{$fd-namespace}-table;
...
+        border-top: $fd-table-border;
+        border-top: $fd-table-border-var;
```
Comment (InnaAtanasova) [inline src/styles/table.scss:295]: same comment as above

**Review by InnaAtanasova — CHANGES_REQUESTED** [review]

File: src/styles/table.scss (hunk 1 above)
Comment (aberikashvili) [reply src/styles/table.scss:14]: done

File: src/styles/table.scss (hunk 2 above)
Comment (aberikashvili) [reply src/styles/table.scss:15]: done

File: src/styles/table.scss (hunk 3 above)
Comment (aberikashvili) [reply src/styles/table.scss:149]: done

File: src/styles/table.scss (hunk 4 above)
Comment (aberikashvili) [reply src/styles/table.scss:154]: done

File: src/styles/table.scss (hunk 5 above)
Comment (aberikashvili) [reply src/styles/table.scss:155]: done

File: src/styles/table.scss (hunk 6 above)
Comment (aberikashvili) [reply src/styles/table.scss:165]: done

File: src/styles/table.scss (hunk 7 above)
Comment (aberikashvili) [reply src/styles/table.scss:172]: done

File: src/styles/table.scss (hunk 10 above)
Comment (aberikashvili) [reply src/styles/table.scss:296]: done

File: src/styles/table.scss (hunk 9 above)
Comment (aberikashvili) [reply src/styles/table.scss:294]: what to handle here? `left` and `right` both are `0`

File: src/styles/table.scss [hunk 11]
```diff
@@ -137,6 +139,38 @@ $block: #{$fd-namespace}-table;
...
+      padding: 1rem;
+      text-align: center;
+      width: 100%;
+
+      @include fd-flex-center() {
+        align-content: center;
+      }
+    }
+
+    &:not(.fd-fixed-height) .fd-table__row:last-of-type {
```
Comment (platon-rov) [inline src/styles/table.scss:153]: `.fd-table__row` => `.#{$block}__row`

File: src/styles/table.scss [hunk 12]
```diff
@@ -137,6 +139,38 @@ $block: #{$fd-namespace}-table;
...
+      .#{block}__cell {
+        border-bottom: 0;
+      }
+    }
+
+    .fd-table__body {
```
Comment (platon-rov) [inline src/styles/table.scss:159]: `.fd-table__body` => `.#{$block}__body`

File: src/styles/table.scss (hunk 7 above)
Comment (platon-rov) [reply src/styles/table.scss:170]: I think as it relates to the table's empty message item it shouldn't be a separate class, let it be modifier for base class, what do you think?

File: src/styles/table.scss (hunk 11 above)
Comment (aberikashvili) [reply src/styles/table.scss:153]: Done

File: src/styles/table.scss (hunk 12 above)
Comment (aberikashvili) [reply src/styles/table.scss:159]: Done

**Comment by aberikashvili** [comment]
> @aberikashvili do we have examples showing those changes?

@droshev I added empty table example, please check again

File: src/styles/table.scss (hunk 7 above)
Comment (platon-rov) [reply src/styles/table.scss:170]: Bump for this one, `.fd-fixed-height` not the correct class name.

**Review by platon-rov — COMMENTED** [review]
Good job, please resolve all conversations :)

**Pull request closed without merging**